runtime/lua/vscode-neovim/api.lua

```diff
@@ -23,6 +23,23 @@ function M.call_extension(command, ...)
     return vim.rpcrequest(vim.g.vscode_channel, plugin_event_name, command, { ... })
 end
 
+-- send command to vscode with range (line or char). [1, 1]-based.
+function M.call_range(command, line1, line2, leaveSelection, ...)
+    return M.call_extension('range-command', command, 'V', line1, line2, 1, 1, leaveSelection, { ... })
+end
+
+function M.notify_range(command, line1, line2, leaveSelection, ...)
+    return M.notify_extension('range-command', command, 'V', line1, line2, 1, 1, leaveSelection, { ... })
```

Comment:
Seems overwrought to have all these dedicated variants. `call()` and `notify()` could cover all these variations with a "kwargs" style dict parameter. That would also simplify the documentation. It also makes the calls, and usage, much easier to understand, instead of having 4+ positional args.

The variadic args `...` could just be a list, there's no benefit to variadic args here: they are hard to visually parse vs the non-variadic args.

```lua
notify('foo', { arg1, arg2, … } {
  line1=…,
  line2=…,
  col1=…,  -- optional, treat as "line" if `nil`
  col2=…,  -- optional, treat as "line" if `nil`
}
```